### PR TITLE
Clean up of the build dog script

### PR DESCRIPTION
**Pull request by vittorione94**
Things that are different:
1. Split build_dog script in different files to increase readability.
2. A floor to the base xml, so that the final model can be ready to use.
3. Overall clean-up of the file, for example the path for the bones was broken I had to remove a ' \ ' symbol.
4. Moved a geometry from extras to parent directory, needed for the final model to compile.

Happy to discuss all the modifications :)

### CODE REVIEW

**Review by yuvaltassa — APPROVED** [review]

**Comment by saran-t** [comment]
Hi, sorry for the late reply -- we haven't got an automatic GitHub->Google thing set up for `dm_control`.

A few things were picked up by our internal linter:
1. New files are missing copyright headers. You can copy these from any of our existing files and change the year to 2022.
2. New files should use 2-space indent with 80-char line limits. Functions should come with docstrings with argument lists (see existing files).
3. New files should have a `"""Module docstring."""` right underneath the copyright header.

**Comment by vittorione94** [comment]
Hi @saran-t,

I've reformatted the files to add all the problems picked up from the linter :)

**Comment by saran-t** [comment]
@vittorione94 can you please rebase to `main`?

**Comment by vittorione94** [comment]
@saran-t done 👍

**Comment by saran-t** [comment]
Vittorio, you merged `main` into your PR branch rather than rebase the PR on top of main. I'll see what I can do on my end but if you get a chance to actually rebase it please do so since Copybara seems a bit confused at the moment.

File: dm_control/locomotion/walkers/assets/dog_v2/build_dog.py [hunk 1]
```diff
@@ -15,43 +15,104 @@
...
-from dm_control.mujoco.wrapper.mjbindings import enums
-import editdistance
-import ipdb  # pylint: disable=unused-import
 from lxml import etree
 
-import numpy as np
-from scipy import spatial
-from scipy.sparse.csgraph import minimum_spanning_tree as mst
-
+import add_torque_actuators
```
Comment (saran-t) [inline dm_control/locomotion/walkers/assets/dog_v2/build_dog.py:25]: Please use absolute imports w.r.t. dm_control.

File: dm_control/locomotion/walkers/assets/dog_v2/build_dog.py [hunk 2]
```diff
@@ -15,43 +15,104 @@
 
 """Make dog model."""
 
-import collections
 import os
-import struct
+
+import numpy as np
```
Comment (saran-t) [inline dm_control/locomotion/walkers/assets/dog_v2/build_dog.py:20]: absl should come before numpy (alphabetical ordering of non-builtin imports)

File: dm_control/locomotion/walkers/assets/dog_v2/build_dog.py [hunk 3]
```diff
@@ -15,43 +15,104 @@
...
+import create_skin
+from dm_control import mjcf
 from dm_control.utils import io as resources
 
 flags.DEFINE_boolean(
-    'make_skin', True, 'Whether to make a new dog_skin.skn')
+  'make_skin', True, 'Whether to make a new dog_skin.skn')
 flags.DEFINE_boolean(
-    'use_tendons', False, 'Whether to add tendons to the model.')
+  'use_tendons', False, 'Whether to add tendons to the model.')
```
Comment (saran-t) [inline dm_control/locomotion/walkers/assets/dog_v2/build_dog.py:38]: Seems unused, please remove.

File: dm_control/locomotion/walkers/assets/dog_v2/build_dog.py [hunk 4]
```diff
@@ -60,15 +121,15 @@ def main(argv):
   # Read flags.
   if FLAGS.is_parsed():
     use_tendons = FLAGS.use_tendons
```
Comment (saran-t) [inline dm_control/locomotion/walkers/assets/dog_v2/build_dog.py:123]: Seems unused, please remove.

**Pull request merged**